x-pack/elastic-agent/control.proto

```diff
@@ -0,0 +1,118 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+syntax = "proto3";
+
+package proto;
+
+option cc_enable_arenas = true;
+option go_package = "pkg/agent/control/proto;proto";
+
+// Status codes for the current state.
+enum Status {
+  STARTING = 0;
+  CONFIGURING = 1;
+  HEALTHY = 2;
+  DEGRADED = 3;
+  FAILED = 4;
+  STOPPING = 5;
+  UPGRADING = 6;
```

Comment:
related to that can we introduce status for ROLLBACK? which takes place in between UPGRADING and FAILED

Reply:
Yes will add rollback